rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -166,12 +170,12 @@ def _enrich_protocol_tranfers(self, context: EnricherContext) -> Optional[str]:
                     f'{context.event.asset}. {str(e)}',
                 )
                 # Don't try other rules since all of them will fail to resolve the asset
-                return None
+                return DEFAULT_ENRICHMENT_OUTPUT
 
-            if transfer_enrich.matched_counterparty is not None:
-                return transfer_enrich.matched_counterparty
+            if transfer_enrich != DEFAULT_ENRICHMENT_OUTPUT:
+                return transfer_enrich
```

Comment:
but this doesn't fix the logic unless you change the `transfer_enrich` in the modules right? it will continue to loop even if the transaction was enriched.

Is either to set the counterparty (lefteris wanted to avoid it) or add a flag `enriched=True` and break if True

Reply:
This change I made to handle the case if `refresh_balances` is true. As for fixing extra running enrichers I was not sure if this needs fixing. But yeah, a flag can be added